components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -125,34 +126,49 @@ func (wsc *WorkspaceController) Reconcile(ctx context.Context, req ctrl.Request)
 	glog.WithField("workspaceID", workspace.Name).WithField("phase", workspace.Status.Phase).Debug("Reconcile workspace")
 
 	if workspace.Status.Phase == workspacev1.WorkspacePhaseCreating ||
-		workspace.Status.Phase == workspacev1.WorkspacePhaseInitializing ||
-		workspace.Status.Phase == workspacev1.WorkspacePhaseRunning {
+		workspace.Status.Phase == workspacev1.WorkspacePhaseInitializing {
 
 		result, err = wsc.handleWorkspaceInit(ctx, &workspace, req)
 		return result, err
 	}
 
 	if workspace.Status.Phase == workspacev1.WorkspacePhaseStopping {
+
 		result, err = wsc.handleWorkspaceStop(ctx, &workspace, req)
 		return result, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
+// latestWorkspace checks if the we have the latest generation of the workspace CR. We do this because
+// the cache could be stale and we retrieve a workspace CR that does not have the content init/backup
+// conditions even though we have set them previously. This will lead to us performing these operations
+// again. To prevent this we wait until we have the latest workspace CR.
+func (wsc *WorkspaceController) latestWorkspace(ctx context.Context, ws *workspacev1.Workspace) bool {
+	ws.Status.SetCondition(workspacev1.NewWorkspaceConditionRefresh())
+
+	err := wsc.Client.Status().Update(ctx, ws)
+	return !errors.IsConflict(err)
```

Comment:
if there's a different error (e.g. api server temporarily unavailable), we'd want to also skip init/dispose I think, as we weren't able to confirm if the resource is the latest version. Maybe return a `(bool, error)` here, where `err` is non-nil on any other error than a conflict error?

Wdyt? Other than that lgtm

Reply:
Wouldn't that just mean that we would want to skip on every error? So we could just return an error and skip if it is non nil.

Reply:
Wouldn't that just mean that we would want to skip on every error? So we could just return an error and skip if it is non nil.